### PR TITLE
SCAL-221973 Implemented CF for Custom-Bar-Chart

### PR DESCRIPTION
**Pull request by yashvardhan-n**

### CODE REVIEW

File: example/custom-bar-chart/custom-chart-plugins.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @file Custom Bar Chart Plugins
+ *
+ * @fileoverview
+ *
+ * @author Yashvardhan Nehra <yashvardhan.nehra@thoughtspot.com>
+ *
+ * Copyright: ThoughtSpot Inc. 2024
+ */
+
+export function createPlotlinePlugin(dataModel) {
```
Comment (harshmeetTS) [inline example/custom-bar-chart/custom-chart-plugins.ts:11]: update readme for this example chart as well

File: example/custom-bar-chart/custom-chart.ts [hunk 2]
```diff
@@ -75,29 +70,41 @@ function getColumnDataModel(
...
+                    type: `${type}`,
+                    backgroundColor: color,
+                    borderColor: color,
+                    datalabels: {
+                        anchor: 'end',
+                    },
+                    plotlines,
+                    plotbands,
+                };
+            }),
```
Comment (harshmeetTS) [inline example/custom-bar-chart/custom-chart.ts:107]: this change removes support for custom styling from the example probably even dynamic changes are we expecting that ?

File: src/main/custom-chart-context.ts [hunk 3]
```diff
@@ -143,6 +143,7 @@ export type CustomChartContextProps = {
 
     // Whether user wants thoughtspot default number and conditional formatting
     allowedConfigurations?: AllowedConfigurations;
+    // batchsize TODO
```
Comment (harshmeetTS) [inline src/main/custom-chart-context.ts:146]: add BatchSize of type number and add comment on top of it that needs to be implemented on TS side that way we would not need to bump the version whenwe eventually support it on TS

File: example/custom-bar-chart/custom-chart-plugins.ts (hunk 1 above)
Comment (yashvardhan-n) [reply example/custom-bar-chart/custom-chart-plugins.ts:23]: ack

File: src/main/custom-chart-context.ts (hunk 3 above)
Comment (yashvardhan-n) [reply src/main/custom-chart-context.ts]: ack

**Review by harshmeetTS — APPROVED** [review]

**Pull request merged**